layer/event_logging.h

```diff
@@ -109,10 +108,16 @@ class Event {
 
   LogLevel GetLogLevel() const { return log_level_; }
 
+ protected:
+  template <typename... Args>
+  void InitAttributes(Args... args) {
```

Comment:
either `void InitAttributes(Args*... args)` to make it clear that we expect pointers
or alternatively, we can get rid of the template with something like:
```c++
void InitAttributes(std::initializer_list<Attribute *> attrs) {
  attributes_ = {attrs.begin(), attrs.end()};
}
```